server/aws-lsp-codewhisperer/src/language-server/agenticChat/tools/chatDb/chatDb.ts

```diff
@@ -509,8 +515,17 @@ export class ChatDatabase {
                     this.#features.logging.error(`Error counting toolResults: ${String(e)}`)
                 }
             }
+            // Count characters in tool spec for the current user message
+            if (message.userInputMessageContext?.tools) {
```

Comment:
need to count differently for current message, `additionalContext, active file and relatedDocuments`

Reply:
Added